catalog/core/catalog-core-standardframework/src/main/java/org/codice/ddf/catalog/resource/download/impl/ResourceDownload.java

```diff
@@ -0,0 +1,169 @@
+
+import org.codice.ddf.catalog.resource.cache.ResourceCacheServiceMBean;
+import org.codice.ddf.catalog.resource.download.DownloadToCacheException;
+import org.codice.ddf.catalog.resource.download.ResourceDownloadMBean;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.impl.ResourceRequestById;
+import ddf.catalog.resource.ResourceNotFoundException;
+import ddf.catalog.resource.ResourceNotSupportedException;
+
+
+public class ResourceDownload implements ResourceDownloadMBean {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ResourceDownload.class);
+
+    private static final String CLASS_NAME = ResourceDownload.class.getName();
+
+    private static final String ERROR_MESSAGE_TEMPLATE = "Unable to download the product associated with metacard [%s] from source [%s] to the product cache.";
+
+    private final CatalogFramework catalogFramework;
+
+    private final ResourceCacheServiceMBean resourceCacheMBean;
+
+    private final MBeanServer mBeanServer;
+
+    private final ObjectName objectName;
+
+    public ResourceDownload(MBeanServer mBeanServer, CatalogFramework catalogFramework,
+            ResourceCacheServiceMBean resourceCacheMBean)
+        throws MalformedObjectNameException {
+        this.mBeanServer = mBeanServer;
+        this.catalogFramework = catalogFramework;
+        this.resourceCacheMBean = resourceCacheMBean;
+        this.objectName = new ObjectName(OBJECT_NAME);
+    }
+
+    public void init() {
+        registerMBean();
+        LOGGER.info("Registered [{}] MBean under object name: [{}].", CLASS_NAME,
+                objectName.toString());
+    }
+
+    public void destroy() {
+        if (objectName != null && mBeanServer != null) {
+            unregisterMBean();
+            LOGGER.info("Unregistered [{}] MBean.", CLASS_NAME);
+        }
+    }
+
+    @Override
+    public String downloadToCache(String sourceId, String metacardId) throws MBeanException {
+        LOGGER.debug(
+                "Downloading resource associated with metacard id [{}] from source [{}] to the product cache.",
+                metacardId, sourceId);
+
+        ResourceRequest resourceRequest = new ResourceRequestById(metacardId);
+
+        if (!resourceCacheMBean.isCacheEnabled()) {
+            String message = "Caching of resources is not enabled.";
+            LOGGER.error(message);
+            throw new MBeanException(new DownloadToCacheException(Status.BAD_REQUEST, message),
+                    message);
+        }
+
+        try {
+            LOGGER.debug(
+                    "Attempting to download the resource associated with metacard [{}] from source [{}] to the product cache.",
+                    metacardId, sourceId);
+            ResourceResponse resourceResponse = catalogFramework.getResource(resourceRequest,
```

Comment:
It is an implementation detail currently that requesting a resource but not reading the stream will fill the cache.  In the future that might not be the case which will break this code.